cmd/github-ldap-user-group-creator/main.go

```diff
@@ -359,24 +363,27 @@ func validate(group *userv1.Group) error {
 	return nil
 }
 
-func UpsertGroup(ctx context.Context, client ctrlruntimeclient.Client, group *userv1.Group) (created bool, err error) {
+func upsertGroup(ctx context.Context, client ctrlruntimeclient.Client, group *userv1.Group) (modified bool, err error) {
 	err = client.Create(ctx, group.DeepCopy())
 	if err == nil {
 		return true, nil
 	}
 	if !errors.IsAlreadyExists(err) {
-		return false, err
+		return false, fmt.Errorf("[1] create failed: %w", err)
 	}
 	existing := &userv1.Group{}
 	if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Name: group.Name}, existing); err != nil {
-		return false, err
+		return false, fmt.Errorf("[2] get failed: %w", err)
 	}
 	if equality.Semantic.DeepEqual(group.Users, existing.Users) {
 		return false, nil
 	}
 	if err := client.Delete(ctx, existing); err != nil {
-		return false, fmt.Errorf("delete failed: %w", err)
+		return false, fmt.Errorf("[3] delete failed: %w", err)
 	}
 	// Recreate counts as "Update"
-	return false, client.Create(ctx, group)
+	if err := client.Create(ctx, group); err != nil {
+		return false, fmt.Errorf("[4] create failed: %w", err)
+	}
+	return true, nil
```

Comment:
This changes the semantics though: from "true only when creating" to "true when creating or modifying".

Reply:
Yeah, but there's no consumer of that value until now, so we can change semantics like we please :smiling_imp: